### PR TITLE
1500 fix lifecycle@devel

### PR DESCRIPTION
**Pull request by bms63**
Thank you for your Pull Request! We have developed this task checklist from the [Development Process Guide](https://pharmaverse.github.io/admiraldev/main/articles/development_process.html) to help with the final steps of the process. Completing the below tasks helps to ensure our reviewers can maximize their time on your code as well as making sure the admiral codebase remains robust and consistent.   

Please check off each taskbox as an acknowledgment that you completed the task or check off that it is not relevant to your Pull Request. This checklist is part of the Github Action workflows and the Pull Request will not be merged into the `devel` branch until you have checked off each task.

- [ ] Place Closes #<insert_issue_number> into the beginning of your Pull Request Title (Use Edit button in top-right if you need to update)
- [ ] Code is formatted according to the [tidyverse style guide](https://style.tidyverse.org/). Run `styler::style_file()` to style R and Rmd files
- [ ] Updated relevant unit tests or have written new unit tests - See [Unit Test Guide](https://pharmaverse.github.io/admiraldev/main/articles/unit_test_guidance.html#writing-unit-tests-in-admiral-)
- [ ] If you removed/replaced any function and/or function parameters, did you fully follow the [deprecation guidance](https://pharmaverse.github.io/admiraldev/main/articles/programming_strategy.html#deprecation-1)?
- [ ] Update to all relevant roxygen headers and examples 
- [ ] Run `devtools::document()` so all `.Rd` files in the `man` folder and the `NAMESPACE` file in the project root are updated appropriately
- [ ] Address any updates needed for vignettes and/or templates
- [x] Update `NEWS.md` if the changes pertain to a user-facing function (i.e. it has an `@export` tag) or documentation aimed at users (rather than developers)
- [ ] Build admiral site `pkgdown::build_site()` and check that all affected examples are displayed correctly and that all new functions occur on the "[Reference](https://pharmaverse.github.io/admiral/reference/index.html)" page. 
- [ ] Address or fix all lintr warnings and errors - `lintr::lint_package()`
- [ ] Run `R CMD check` locally and address all errors and warnings - `devtools::check()`
- [x] Link the issue in the Development Section on the right hand side.
- [ ] Address all merge conflicts and resolve appropriately 
- [x] Pat yourself on the back for a job well done! Much love to your accomplishment!

### CODE REVIEW

File: tests/testthat/test-deprecation.R [hunk 1]
```diff
@@ -185,28 +172,24 @@ test_that("date_source: errors when preserve is specified", {
...
 
 test_that("derive_var_agegr_ema Test 1: A warning is issued if `derive_var_agegr_ema()` is called", { # nolint
-  with_options(lifecycle_verbosity = "warning", {
+  rlang::with_options(lifecycle_verbosity = "warning", {
     expect_warning(
-      derive_var_agegr_ema(admiral_dm, age_var = AGE, new_var = AGEGR1),
+      derive_var_agegr_ema(admiral.test::admiral_dm, age_var = AGE, new_var = AGEGR1),
       "deprecated",
       fixed = TRUE
     )
```
Comment (thomas-neitmann) [inline tests/testthat/test-deprecation.R:185]:
```suggestion
      derive_var_agegr_ema(admiral.test::admiral_dm, age_var = AGE, new_var = AGEGR1),
      class = "lifecycle_warning_dprecated"
    )
```

File: tests/testthat/test-deprecation.R [hunk 2]
```diff
@@ -185,28 +172,24 @@ test_that("date_source: errors when preserve is specified", {
...
 })
 
 test_that("derive_var_agegr_fda Test 1: A warning is issued if `derive_var_agegr_fda()` is called", { # nolint
-  with_options(lifecycle_verbosity = "warning", {
+  rlang::with_options(lifecycle_verbosity = "warning", {
     expect_warning(
-      derive_var_agegr_fda(admiral_dm, age_var = AGE, new_var = AGEGR1),
+      derive_var_agegr_fda(admiral.test::admiral_dm, age_var = AGE, new_var = AGEGR1),
       "deprecated",
       fixed = TRUE
```
Comment (thomas-neitmann) [inline tests/testthat/test-deprecation.R:194]:
```suggestion
      derive_var_agegr_fda(admiral.test::admiral_dm, age_var = AGE, new_var = AGEGR1),
      class = "lifecycle_warning_deprecated"
```

**Review by thomas-neitmann — CHANGES_REQUESTED** [review]
There's still a lot of test cases which either only check for a regular expression or have both the `class` and regular expression including the `fixed` argument set. We should only specify `class` and remove the regular expressions completly.

**Comment by thomas-neitmann** [comment]
Also @bms63 please assign yourself to the issue next time. I actually had someone from the community who wanted to pick this up but then saw your PR.

**Comment by bms63** [comment]
Hey @thomas-neitmann and @bundfussr.  Do you think we could get this merged in and sent off to CRAN?  I'd like to merge this into `devel` as I think this will cause some issues with our Action that tests the latest package releases, i.e. {lifecycle}.

**Review by bundfussr — APPROVED** [review]

**Review by thomas-neitmann — APPROVED** [review]

**Pull request merged**